temporalio/ui-server
# Support pluggable config providers

## What was changed


Adds support for pluggable Config providers:

* adds Config Provider interface
* it's wrapped in a Config Provider With Refresh (that periodically gets the config and caches it)
* there is a default implementation of Config Provider that is used in our docker image (fs_config_provider)
* people can start the server with their own provider instead  

## Why?


Flexibility in how the config is provided
TLS params periodic refresh

## Checklist


1. Closes 

2. How was this tested:


* First start Temporal service with mTLS disabled
* In UI server's config set `refreshInterval: 30s`, ex in development.yaml file
* Load** UI server and open http://localhost:8080/ and verify namespaces are loaded
* Stop Temporal service
* Start a new Temporal service with mTLS enabled using https://github.com/temporalio/samples-server/tree/main/tls/tls-simple
* Go to UI server development.yaml and set values for `caFile`, `certFile`, `keyFile`, `enableHostVerification`, `serverName`
* Wait up to 30 seconds (refreshInterval) until you see `2022[/02/16]() 11:38:06 loaded new UI server configuration` in the logs, should also print that it loaded new TLS CA, KEY etc
* Open  http://localhost:8080/ again and verify that the namespaces are loaded

3. Any docs updates needed?
